wmfdata/__init__.py

```diff
@@ -9,21 +11,33 @@
     spark
 )
 
+# Warn about the conda-analytics migration once it has almost certainly begun
+if conda.is_anaconda_wmf_env() and date.today() >= date(2022, 12, 12):
+    utils.print_err(
+        "You are using an anaconda-wmf environment.",
+        "You must switch to a conda-analytics environment by 31 March 2023.",
+        "For details, see https://wikitech.wikimedia.org/wiki/Analytics/Systems/Conda."
+    )
+
 try:
     remote = utils.check_remote_version(metadata.source, metadata.version)
+
     if remote['is_newer']:
-        update_message = (
-            "You are using wmfdata v{0}, but v{1} is available.\n\n" +
-            "To update, run `pip install --upgrade git+{2}.git@release --ignore-installed`.\n\n" +
-            "To see the changes, refer to {2}/blob/release/CHANGELOG.md"
-        ).format(metadata.version, remote['version'], metadata.source)
-        utils.print_err(update_message)
+        update_command = f"pip install --upgrade git+{metadata.source}.git@release"
+        if conda.is_anaconda_wmf_env():
+            update_command += " --ignore-installed"
+
+        utils.print_err(
+            f"You are using Wmfdata v{metadata.version}, but v{remote['version']} is available.",
+            f"To update, run `{update_command}`.",
+            f"To see the changes, refer to {metadata.source}/blob/release/CHANGELOG.md.",
+        )
```

Comment:
Could we modify `check_remote_version()` to also calculate whether the newer version has breaking changes, and include that as part of the notice? That way the potential impact on the user is more explicit, especially if they don't know about semantic versioning.

Reply:
I actually had the same idea, but thought it might be over-engineering! Since you had the same idea, that suggests it's not. I just added it. 

For what it's worth, though, in the long term I think we should drop this homegrown upgrade system and rely on Conda's normal upgrade mechanics. It will mean not prompting users about new versions but I think a reasonable trade-off for just doing things the standard way.